### PR TITLE
Adds Option Select component

### PR DESCRIPTION
**Pull request by domoscargin**
https://trello.com/c/VCANsToa/233-3-replace-search-filters-with-govuk-checkbox-groups-option-select

The Option Select component is heavily based on the Finder Frontend component: https://github.com/alphagov/finder-frontend/blob/master/app/views/components/_option-select.html.erb

It creates a collapsible container with checkboxes. If checkboxes are selected, a count is displayed and updated. We don't include the search text input from Finder.

Test this in operation by running this branch of the Buyer frontend locally: https://github.com/alphagov/digitalmarketplace-buyer-frontend/tree/bk-replace-filters-with-checkboxes 

### Option select on Buyer search pages:
Screenshot 2020-09-24 at 08 02 43

### CODE REVIEW

**Comment by lfdebrux** [comment]
### My review checklist

I've:

- [ ] looked at the code
- [ ] tested locally

File: src/digitalmarketplace/components/option-select/option-select.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    height = 200
+  }
+
+  // Resize if the list is only slightly bigger than its container
+  if (height < initialOptionContainerHeight + 50) {
+    this.setContainerHeight(height + 1)
+    return
+  }
+
+  // Resize to cut last item cleanly in half
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/option-select/option-select.js:170]: This made me actually gasp at it's audacity. Well done!

File: src/digitalmarketplace/components/option-select/option-select.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  required: true
+  description: The title of the option select
+- name: name
+  type: string
+  required: true
+  description: The name of the option select element
+- name: options_container_id
+  type: string
+  required: true
+  description: The ID of the container for the Checkboxes component
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/option-select/option-select.yaml:13]: Why is this needed?

File: src/digitalmarketplace/components/option-select/option-select.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+- name: items
+  type: array
+  required: true
+  description: The checkbox items
+- name: closed_on_load
+  type: boolean
+  description: Whether to collapse the filter by default
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/option-select/option-select.yaml:20]: If this is optional, what is the default value?

**Review by lfdebrux — APPROVED** [review]
This looks really great, I love it! 😍

File: src/digitalmarketplace/components/option-select/option-select.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  required: true
+  description: The name of the option select element
+- name: options_container_id
+  type: string
+  required: true
+  description: The ID of the container for the Checkboxes component
+- name: items
+  type: array
+  required: true
+  description: The checkbox items
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/option-select/option-select.yaml:17]: Can we document what an item looks like?

Comment (lfdebrux) [reply src/digitalmarketplace/components/option-select/option-select.yaml:17]: (We could probably just put a link to checkboxes item docs https://design-system.service.gov.uk/components/checkboxes/#options-checkboxes-example--items).

File: src/digitalmarketplace/components/option-select/option-select.yaml (hunk 2 above)
Comment (domoscargin) [reply src/digitalmarketplace/components/option-select/option-select.yaml:13]: Finder uses it for querying and because they have the searchbox in the same parent container. 

DMp has it in place on the old option boxes, and I figured it provides an easy route to the checkboxes component, and there might be some obscure use of the ID somewhere in a backend function that I'd mess up if I removed it!

File: src/digitalmarketplace/components/option-select/option-select.yaml (hunk 3 above)
Comment (domoscargin) [reply src/digitalmarketplace/components/option-select/option-select.yaml]: Good spot, popped a default into the docs
